### ISSUE TITLE
No traces being sent to Jaeger

### ISSUE THREAD

**Issue by robertmarsal** [benthosdev/benthos#1314]
Hello,

I am trying to send traces from Benthos to Jaeger using v4 however nothing gets sent. The same similar config seems to work with v3 but nothing happens in v4.

Below you can find a way to reproduce the issue:

Given a `docker-compose.yaml` file:
```yaml
services:
    jaeger:
      container_name: jaeger
      image: jaegertracing/all-in-one:1.25
      ports:
        - '16686:16686'
      networks:
        test:

    benthos:
      container_name: benthos
      image: jeffail/benthos:4.3
      depends_on:
        - jaeger
      command:
        - -c
        - /test.yaml
      volumes:
        - ./test.yaml:/test.yaml:ro
      ports:
        - '4195:4195'
      networks:
        test:

networks:
  test:

```

And a `test.yaml` file:
```yaml
logger:
  level: ALL

tracer:
  jaeger:
    agent_address: jaeger:6831
    sampler_type: const
    sampler_param: 1
    flush_interval: "1s"

input:
  http_server:
    path: /

output:
  stdout:
    codec: lines
```

When I make an HTTP request to `http://localhost:4195/` I can see the output in the container logs but I can't see any traces in Jaeger.

```HTTP
POST http://localhost:4195/
Content-Type: application/json

{"Hello":  "World"}
```

Logs:
```
benthos    | level=info msg="Running main config from specified file" @service=benthos path=/test.yaml
benthos    | level=info msg="Launching a benthos instance, use CTRL+C to close" @service=benthos
benthos    | level=info msg="Listening for HTTP requests at: http://0.0.0.0:4195" @service=benthos
benthos    | level=trace msg="Consumed 1 messages from POST to '/'." @service=benthos label="" path=root.input
benthos    | level=trace msg="Attempting to write 1 messages to 'stdout'." @service=benthos label="" path=root.output
benthos    | {"Hello":  "World"}
benthos    | level=trace msg="Successfully wrote 1 messages to 'stdout'." @service=benthos label="" path=root.output
```

Using made up urls for the agent doesn't seem to cause any logs or errors, there's also no log on startup to indicate tracing is active.

**Comment by Jeffail** [comment]
Hey @robertmarsal, unfortunately it doesn't look as though the open telemetry APIs gives us any information about the availability of the agent, so connectivity problems will be difficult to diagnose. I can definitely see traces when I run it all locally but if it specifically stopped working for you after V4 that suggests maybe something broke after the open telemetry migration. Is it the case that moving back to V3 with your example set up shows traces coming back?

**Comment by robertmarsal** [comment]
Yes, in the example above switching the image tag from 4.3 to 3 makes it work again

**Comment by robertmarsal** [comment]
Adding a few more details to this. After some investigation I noticed that if I run benthos from source (go run) it will send the traces to Jaeger, however when running inside a Docker container (with the official) image, traces stop being sent. 

My suspicion at the moment is this busybox bug: https://bugs.busybox.net/show_bug.cgi?id=14671 
(more info here: https://github.com/kubernetes/kubernetes/issues/66924) however after I downgraded busybox to 1.28.0 I am now seeing `UDP connection not yet initialized, an address has not been resolved` in the logs.

It works from the Docker container if I use the IP of Jaeger however it doesn't work with a DNS name
